sdk/python/kfp/compiler/main.py

```diff
@@ -97,9 +101,19 @@ def main():
   args = parse_arguments()
   if args.py is None:
     raise ValueError('The --py option must be specified.')
+  mode = None
+  # when args.mode is not provided, pass mode=None to let compiler handle the default value
+  if args.mode:
+    if args.mode == 'V2_COMPATIBLE':
+      mode = dsl.PipelineExecutionMode.V2_COMPATIBLE
+    elif args.mode == 'V1_LEGACY' or args.mode == 'V1':
```

Comment:
> `or args.mode == 'V1'`

For consistence's matter, we should either disallow this here, or allow it for the environment variable approach as well (which current raises a ValueError on this).

Reply:
That's a good catch! Let me fix this

Reply:
Updated, now we are consistent that 'V1' and 'V1_LEGACY' are both supported, while in user helper messages, we only show people 'V1', because 'V1' has not been deprecated.